src/main/java/delete_post/interface_adapters/DeletePostPresenter.java

```diff
@@ -2,20 +2,22 @@
 
 import delete_post.use_case.DeletePostOutputBoundary;
 import delete_post.use_case.DeletePostResponseModel;
-import entities.User;
+import delete_post.UI.DeleteErrorView;
 
 public class DeletePostPresenter implements DeletePostOutputBoundary{
 
     private final DeletePostViewModel viewModel;
+    private final Jframe postView;
 
     public DeletePostPresenter(DeletePostViewModel viewModel){
         this.viewModel = viewModel;
     }
     @Override
-    public void prepareFailView(String error){
-        //error popup, no change to posts view or database;
+    public void prepareFailView(){
+        DeleteErrorView(postView);
```

Comment:
I don't think you should be calling the error view directly from the presenter. Rather, you should add a method to the viewmodel and use the observer design pattern